### DISCUSSION TITLE
Psyko pypi package; Taiko with Python?

### DISCUSSION THREAD

**Discussion by jaydee713** [getgauge/taiko#2065] [General]
I realize there are a multitude of responses indicating Taiko is JS-only... just wondering if @BugDiver's work for the pypi package "psyko" was experimental then, and I shouldn't actually try to use it? :)

https://pypi.org/project/psyko/

As a sub-note, what's the main reason for why Taiko isn't compatible with Python? Is it an underlying async issue, or just resources are too limited to support it?

**Comment by bugdiver** [comment]
@jaydee713 The python port for taiko was an experiment, and It could never make it to production-grade. Currently, Taiko is only available in JS.

**Comment by zabil** [comment]
> As a sub-note, what's the main reason for why Taiko isn't compatible with Python? Is it an underlying async issue, or just resources are too limited to support it?

Taiko uses [CRI](https://github.com/cyrus-and/chrome-remote-interface) a javascript library to automate Chrome (and other [CDP](https://chromedevtools.github.io/devtools-protocol/) compatible browsers). At the moment there are no official or well maintained python CDP clients.
